tensorflow/core/kernels/tile_ops.cc

```diff
@@ -188,6 +189,18 @@ class TileOp : public OpKernel {
           context, multiples_array[i] >= 0,
           errors::InvalidArgument("Expected multiples[", i, "] >= 0, but got ",
                                   multiples_array[i]));
+      if (MultiplyWithoutOverflow(
+          output_shape.num_elements(), multiples_array[i]) < 0) {
+        string msg;
+        for (int ii = 0; ii < input_dims; ++ii) {
+          if (ii != 0) {
+            strings::StrAppend(&msg, ", ");
+          }
+          strings::StrAppend(&msg, multiples_array[ii]);
+        }
+        OP_REQUIRES(context, false, errors::InvalidArgument(
+            "Shape [", msg, "] has too many elements"));
+      }
       output_shape.AddDim(input.dim_size(i) * multiples_array[i]);
```

Comment:
Let's instead change this to use [`AddDimWithStatus`](https://github.com/tensorflow/tensorflow/blob/51d911b4661b290f0e207fc7cf1ec79b9e411a58/tensorflow/core/framework/tensor_shape.h#L208-L210) and wrap that in `OP_REQUIRES_OK`.

Also, if we want to display a shape in an error message, `shape.DebugString()` is the best API to use.